### DISCUSSION TITLE
Options for Booster and kernels?

### DISCUSSION THREAD

**Discussion by abenz1267** [archlinux/archinstall#512] [Ideas]
Hi,

would it be within scope to add options to use booster instead of mkinitcpio and linux-zen over linux? Does that make sense? Also: what about intel/amd ucode?

Regards

**Comment by Torxed** [comment]
Ucode support has been in for a while and will come in the next release: https://github.com/archlinux/archinstall/pull/221
As for linux-zen and other kernels, has also been added about a month ago: https://github.com/archlinux/archinstall/pull/315

As for booster however, I'm not sure. mkinitcpio is what's officially used to build the ISO's (unless something's changed very recently) and I kinda like that it's pretty stable for what we're doing. I don't mind exploring the idea of using something else, or at least making us agnostic to whatever we want to use. It would more or less just be one more function or change the existing function: https://github.com/archlinux/archinstall/blob/7daaf1143fca2723bfb63ab4e3030485446da1c0/archinstall/lib/installer.py#L294-L300

But I feel that it's outside of the scope here and I don't know what risks/issues that might cause.

**Reply by Torxed** [reply]
I'll add to this, during development of https://github.com/archlinux/archinstall/pull/393 I figured one of the hooks might be able to do this. If you write a plugin, and load it (this would require a custom installation script) with a `on_mkinitcpio()` hook that returns anything *(for instance just a `return True`)* - you could interrupt the use of `mkinitcpio` and call whatever you like.

This is not as smooth and easy as the guided installer, but this would allow you to do this very thing.
Assuming that PR gets included.
